packages/stark-build/config/webpack.common.js

```diff
@@ -465,18 +468,17 @@ module.exports = function(options) {
 				},
 				platform: 0, // 0 = browser, 1 = server
 				compilerOptions: {
-					// FIXME Those options force AngularCompilerOptions. Cannot be overwritten by angularCompilerOptions defined in tsconfig.json
-					generateCodeForLibraries: true,
-					skipMetadataEmit: false,
-					strictMetadataEmit: false,
-					skipTemplateCodegen: false,
-					strictInjectionParameters: true,
-					fullTemplateTypeCheck: true,
-					annotateForClosureCompiler: false,
-					annotationsAs: "static fields",
-					enableLegacyTemplate: false,
-					preserveWhitespaces: false,
-					allowEmptyCodegenFiles: false,
+					generateCodeForLibraries: typeof appNgcOptions.generateCodeForLibraries === "undefined" ? true : appNgcOptions.generateCodeForLibraries,
```

Comment:
We could improve this to make it cleaner and easier to mantain. What about this?:

```
...
const defaultNgcOptions = {
    "generateCodeForLibraries": true,
    "skipMetadataEmit": false,
    "strictMetadataEmit": false,
    "skipTemplateCodegen": false,
    "strictInjectionParameters": true,
    "fullTemplateTypeCheck": true,
    "annotateForClosureCompiler": false,
    "annotationsAs": "static fields",
    "enableLegacyTemplate": false,
    "preserveWhitespaces": false,
    "allowEmptyCodegenFiles": false
}

const appNgcOptions = Object.assign(
   {},
   defaultNgcOptions,
   tsConfigApp.raw.angularCompilerOptions,
   module: !isProd && METADATA.WATCH ? "commonjs" : "es2015" // TODO is it needed in our case? => Force commonjs module format for TS on dev watch builds.
);

// then in the AngularCompilerPlugin
  ...
      new AngularCompilerPlugin({
				sourceMap: true, // TODO: apply based on tsConfig value?
				mainPath: entry.main,
				skipCodeGeneration: !METADATA.AOT,
				hostReplacementPaths: {
					[helpers.root("src/environments/environment.ts")]: environment
				},
				platform: 0, // 0 = browser, 1 = server
				compilerOptions: appNgcOptions, // simply pass the object
                                tsConfigPath: METADATA.tsConfigPath
`

Reply:
Good idea :)